### DISCUSSION TITLE
texlap configuration

### DISCUSSION THREAD

**Discussion by firubat** [LazyVim/LazyVim#2523] [Q&A] [answered]
[typo in title, should be "texlab configuration"]

I'm trying to add `ignoredPatterns` to texlab configuration, following [this discussion](https://github.com/latex-lsp/texlab/discussions/1004). 
I tried adding the suggested lua table in the lua file were I configure texlab as shown [here](https://www.lazyvim.org/extras/lang/tex) in the file `~/.config/nvim/lua/plugins/vimtex.lua`:
```
  {
    "neovim/nvim-lspconfig",
    optional = true,
    opts = {
      servers = {
        texlab = {
          keys = {
            { "<Leader>K", "<plug>(vimtex-doc-package)", desc = "Vimtex Docs", silent = true },
          },
          diagnostics = {
            ignoredPatterns = { "Unused label", "Undefined reference" },
          },
        },
      },
    },
  },

```
But it doesn't work (texlab works, the patterns aren't ignored). Tried both just `:LspRestart` and also restarting neovim.
Am I putting the config in the right place? Do I have some error? (no errors are shown)

**Comment by dpetka2001** [comment, marked as answer]
Try using `opts.servers.texlab.settings.texlab.diagnostics.ignoredPatterns` instead. Change that to the Lua table representation like also mentioned in the other issue you linked in the texlab repo.

**Reply by firubat** [reply]
OK this seems to work, but only after restart. Thanks!
```
{
    "neovim/nvim-lspconfig",
    optional = true,
    opts = {
      servers = {
        texlab = {
          keys = {
            { "<Leader>K", "<plug>(vimtex-doc-package)", desc = "Vimtex Docs", silent = true },
          },
          settings = {
            texlab = {
              diagnostics = {
                ignoredPatterns = { "Unused label", "Undefined reference" },
              },
            },
          },
        },
      },
    },
  },
```

**Reply by NeilGirdhar** [reply]
If you're using Mason, you can do:
```
        lspconfig.texlab.setup {
            on_attach = on_attach,
            capabilities = capabilities,
            settings = {
                texlab = {
                    diagnostics = {
                        ignoredPatterns = { "Unused label", "Undefined reference" },
                    },
                },
            },
        }
```
